src/main/java/org/snomed/snowstorm/ecl/ECLQueryService.java

```diff
@@ -66,6 +67,17 @@
 		});
 	}
 
+	public Boolean hasAnyResults (String ecl, String branch, BranchCriteria branchCriteria, boolean stated, Collection<Long> conceptIdFilter) {
+		
+		SExpressionConstraint expressionConstraint = (SExpressionConstraint) eclQueryBuilder.createQuery(ecl);
+		Optional<Page<Long>> pageOptional = expressionConstraint.select(branch, branchCriteria, stated, conceptIdFilter, ControllerHelper.getPageRequest(0,1), queryService);
+		
+		if(pageOptional.get().hasNext()) {
```

Comment:
My understanding is that the `hasNext()` method will return true if there is another page of results after the first page. This means that if there is 1 query result the method will return false.

Reply:
It would be cleaner to return the result of this boolean expression in one line, rather than wrapping it in an `if` block and returning `true` or `false`.

Reply:
Each return only ever includes one page due to "getPageRequest(0,1)", but agreed the hasNext is less than ideal.  Switching to hasContent(), and removing the if-block as described.